presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2689,6 +2691,28 @@ private static OptionalInt min(OptionalInt left, OptionalInt right)
         return OptionalInt.of(Math.min(left.getAsInt(), right.getAsInt()));
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layout)
+    {
+        if (!layout.isPresent()) {
+            return;
+        }
+        Preconditions.checkArgument(layout.get() instanceof HiveTableLayoutHandle, "Unexpected table layout type for Hive");
+        HiveTableLayoutHandle hiveLayout = (HiveTableLayoutHandle) layout.get();
+        if (!hiveLayout.getPartitions().isPresent()) {
+            return;
+        }
+        int partitionCount = hiveLayout.getPartitions().get().size();
+        int maxAllowedPartition = HiveSessionProperties.getMaxPartitionsPerScan(session);
+        if (partitionCount > maxAllowedPartition) {
+            throw new PrestoException(HIVE_EXCEEDED_PARTITION_LIMIT, format(
+                    "Query over table '%s' trying to can access %s partitions, more than allowed limit of %s partitions",
```

Comment:
Query ... trying to access...
do we need the `can`?

Reply:
Good catch! updated